From b533ddf4fac14b9583151a2b67984fde0623ee24 Mon Sep 17 00:00:00 2001
From: Konstantin Tokarev <annulen@yandex.ru>
Date: Tue, 17 Oct 2017 22:46:22 +0300
Subject: [cmake] Don't require match of Qt patch release with installed
 QtWebKit version

Allows to use e.g. QtWebKit built against Qt 5.9.0 with Qt 5.9.2

We use private APIs of Qt which have no ABI compatibility guarantees,
however within patch release of same minor release it's highly unlikely
that such changes will be introduced. Sometimes we cannot use tagged Qt version
for Coin builds and in fact use very close commit of qt5 repository,
which nevertheless has different patch version.

Change-Id: I2f8c9abc47cf42693e1556d85f818ad4ec9baa61

diff --git a/Source/PlatformQt.cmake b/Source/PlatformQt.cmake
index 4685caced6f..7f0a542726b 100644
--- a/Source/PlatformQt.cmake
+++ b/Source/PlatformQt.cmake
@@ -74,6 +74,19 @@ target_include_directories(WebKitWidgets INTERFACE
     $<INSTALL_INTERFACE:${KDE_INSTALL_INCLUDEDIR}/QtWebKitWidgets>
 )
 
+set(QTWEBKIT_PACKAGE_INIT "
+macro(find_dependency_with_major_and_minor _dep _major _minor)
+    find_dependency(\${_dep} \"\${_major}.\${_minor}\")
+    if (NOT(\"\${\${_dep}_VERSION_MAJOR}\" STREQUAL \"\${_major}\" AND \"\${\${_dep}_VERSION_MINOR}\" STREQUAL \"\${_minor}\"))
+        set(\${CMAKE_FIND_PACKAGE_NAME}_NOT_FOUND_MESSAGE \"\${CMAKE_FIND_PACKAGE_NAME} could not be found because dependency \${dep} is required to have exact version \${_major}.\${_minor}.x.\")
+        set(\${CMAKE_FIND_PACKAGE_NAME}_FOUND False)
+        return()
+    endif ()
+endmacro ()
+
+####################################################################################
+")
+
 set(_package_footer_template "
 ####### Expanded from QTWEBKIT_PACKAGE_FOOTER variable #######
 
diff --git a/Source/Qt5WebKitConfig.cmake.in b/Source/Qt5WebKitConfig.cmake.in
index 34e9aadcac5..a50d0353fd5 100644
--- a/Source/Qt5WebKitConfig.cmake.in
+++ b/Source/Qt5WebKitConfig.cmake.in
@@ -1,8 +1,9 @@
 @PACKAGE_INIT@
+@QTWEBKIT_PACKAGE_INIT@
 
-find_dependency(Qt5Core    @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Gui     @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Network @Qt5_VERSION@ EXACT)
+find_dependency_with_major_and_minor(Qt5Core @Qt5_VERSION_MAJOR@ @Qt5_VERSION_MINOR@)
+find_dependency_with_major_and_minor(Qt5Gui @Qt5_VERSION_MAJOR@ @Qt5_VERSION_MINOR@)
+find_dependency_with_major_and_minor(Qt5Network @Qt5_VERSION_MAJOR@ @Qt5_VERSION_MINOR@)
 
 include("${CMAKE_CURRENT_LIST_DIR}/WebKitTargets.cmake")
 
diff --git a/Source/Qt5WebKitWidgetsConfig.cmake.in b/Source/Qt5WebKitWidgetsConfig.cmake.in
index ade62eeaed0..ef1c42956f4 100644
--- a/Source/Qt5WebKitWidgetsConfig.cmake.in
+++ b/Source/Qt5WebKitWidgetsConfig.cmake.in
@@ -1,11 +1,8 @@
 @PACKAGE_INIT@
-
-find_dependency(Qt5Core    @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Gui     @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Network @Qt5_VERSION@ EXACT)
-find_dependency(Qt5Widgets @Qt5_VERSION@ EXACT)
+@QTWEBKIT_PACKAGE_INIT@
 
 find_dependency(Qt5WebKit @PROJECT_VERSION_STRING@ EXACT)
+find_dependency_with_major_and_minor(Qt5Widgets @Qt5_VERSION_MAJOR@ @Qt5_VERSION_MINOR@)
 
 include("${CMAKE_CURRENT_LIST_DIR}/Qt5WebKitWidgetsTargets.cmake")
 
